app/uk/gov/hmrc/apiplatformoutboundsoap/services/OutboundService.scala

```diff
@@ -67,14 +68,36 @@ class OutboundService @Inject()(outboundConnector: OutboundConnector, wsSecurity
   private def addHeaders(message: MessageRequest, operation: Operation, envelope: SOAPEnvelope): Unit = {
     val wsaNs: OMNamespace = getOMFactory.createOMNamespace(WSA_NAMESPACE, "wsa")
     addSoapAction(operation, wsaNs, envelope)
+    addMessageHeader(message, operation, envelope)
     addOptionalAddressingHeaders(message, wsaNs, envelope)
   }
 
-  private def addSoapAction(operation: Operation, wsaNs: OMNamespace, envelope: SOAPEnvelope): Unit = {
-    (operation.getInput.getExtensionAttribute(new QName(WSAW_NAMESPACE, "action")) match {
+  private def addMessageHeader(message: MessageRequest, operation: Operation, envelope: SOAPEnvelope): Unit = {
+    val ccnmNamespace: OMNamespace = getOMFactory.createOMNamespace("http://ccn2.ec.eu/CCN2.Service.Platform.Common.Schema", "ccnm")
+    val messageHeader: OMElement = getOMFactory.createOMElement("MessageHeader", ccnmNamespace)
+    envelope.getHeader.addChild(messageHeader)
+
+    def addToMessageHeader(elementName: String, elementContent: String) = {
+      val version = getOMFactory.createOMElement(elementName, ccnmNamespace)
```

Comment:
Oh we forgot to rename this value. It shouldn't be version, but something more generic, like "element". Version was the first one we created